solr/core/src/java/org/apache/solr/cloud/OverseerSolrResponse.java

```diff
@@ -26,7 +26,9 @@
 import java.util.Objects;
 
 public class OverseerSolrResponse extends SolrResponse {
-  
+ 
+  private static final long serialVersionUID = 4721653044098960880L;
```

Comment:
My understanding is that this number can actually vary depending on the compiler, so setting it to a specific value like this (expecting it to be number the class had in previous versions) may not work for everyone. 
Since the changes done in SOLR-14095 are just addition of static methods, maybe the better solution is to just revert them from OverseerSolrResponse and put them in some other util class

Reply:
hi Tomas, that's my understanding too - and it's possible that people may be running builds of earlier Solr versions whose serialVersionUID for this class differ from 472165... . 

I've some (slightly shaky!) evidence in support of doing it this way: in https://github.com/apache/lucene-solr/pull/1140 for SOLR-14165 I saw that the same earlier UID for SolrResponse had been generated by several different stacks, but it's possible the stacks weren't sufficiently different. Also I can see about a dozen instances of serialVersionUID being set to explicit values (other than 1) in the Lucene/Solr codebase, presumably for reasons similar to the current one - though to be fair I've no way to tell if these have caused compatibility issues with custom builds in the past.

I do think that serialVersionUID should be set explicitly for all serializable classes, and that setting it to the value used in the earlier official release builds is the best value to use when it's being set retrospectively like this.

In the interests of finding other approaches to making the serialization vs javabin change backwards-compatible, I've just run a build that didn't set serialVersionUID explicitly but made useUnsafeSerialization and useUnsafeDeserialization private, to see if this would give me the same default UID as before. It didn't - I got a third value -550706... instead. So as far as I can see the only options here are to remove all the changes from OverseerSolrResponse and put them elsewhere as you describe above, or to set its serialVersionUID to a possibly-incompatible value as I've done in this PR.

hope this helps!
Andy

Reply:
I agree, everything that uses java serialization should be setting a serialVersionUID, my concern is that it may be too late now. I think you discovered a bug with your test (thanks!), but I believe it's too late to add a serialVersionUID now because for some users it could mean exactly the same as if we'd had one before and now we are changing it.
Hopefully this won't be an issue once the serialization is in javabin and the Java serialization part is removed.

Reply:
hi Tomas,

I still think it'd be better to set the serialVersionUID. It's _possible_ (though I think unlikely*) that there are systems where the previous (computed) value is different to `472165...`, but the new computed value (with the change to the class) would be different anyway, so either way they'll see an incompatibility. On systems using the standard build, we can make the new class backwards-compatible by adding serialVersionUID. It's guaranteed to be incompatible for everyone if we don't.

Andy

\* My reading of https://docs.oracle.com/javase/7/docs/platform/serialization/spec/class.html is that the computed UID is independent of the Java version, and that it should be set to the previously-computed value in version 2+ of a class.

Reply:
> It's possible (though I think unlikely*) that there are systems where the previous (computed) value is different to 472165..., but the new computed value (with the change to the class) would be different anyway

That's why I suggest that we revert the changes to the class

> On systems using the standard build, we can make the new class backwards-compatible by adding serialVersionUID. It's guaranteed to be incompatible for everyone if we don't.

But removing the changes in OverseerSolrResponse also fixes the compatibility issue, for all cases in theory. For versions higher than 8.5, the solution will be to move to Javabin. @andywebb1975, I proposed the other approach in https://github.com/apache/lucene-solr/pull/1227. mind taking a look?